server/sonar-server/src/main/java/org/sonar/server/platform/ServerFileSystemImpl.java

```diff
@@ -38,10 +36,10 @@
   private final File dataDir;
   private final File deployDir;
 
-  public ServerFileSystemImpl(Settings settings) {
-    this.homeDir = new File(requireNonNull(settings.getString(ProcessProperties.PATH_HOME)));
-    this.tempDir = new File(requireNonNull(settings.getString(ProcessProperties.PATH_TEMP)));
-    this.dataDir = new File(requireNonNull(settings.getString(ProcessProperties.PATH_DATA)));
+  public ServerFileSystemImpl(Configuration config) {
+    this.homeDir = new File(config.get(ProcessProperties.PATH_HOME).get());
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Call "Optional#isPresent()" before accessing the value. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS3655)